library/src/helpers/__tests__/schema.test.ts

```diff
@@ -423,4 +397,205 @@ describe('SchemaHelpers', () => {
+          bar: { type: 'string' },
+          zor: { type: 'string' },
+        },
+        dependencies: {
+          foo: ['bar'],
+          bar: ['foo', 'zor'],
+          zor: ['foo', 'bar'],
+        },
+      });
+
+      const fooResult = SchemaHelpers.getDependentRequired('foo', schema);
+      expect(fooResult).toEqual(['bar', 'zor']);
+      const barResult = SchemaHelpers.getDependentRequired('bar', schema);
+      expect(barResult).toEqual(['foo', 'zor']);
+      const zorResult = SchemaHelpers.getDependentRequired('zor', schema);
+      expect(zorResult).toEqual(['bar']);
+    });
+  });
+
+  describe('.getDependentSchemas', () => {
+    test('should return undefined when dependencies is not defined', () => {
+      const schema = new Schema({
+        properties: {
+          type: 'object',
+          properties: {
+            name: {
+              type: 'string',
+            },
+            credit_card: {
+              type: 'string',
+            },
+          },
+          required: ['name'],
+        },
+      });
+
+      const result = SchemaHelpers.getDependentSchemas(schema);
+      expect(result).toEqual(undefined);
+    });
+
+    test('should return dependent schemas (simple case)', () => {
+      const jsonSchema = {
+        type: 'object',
+        properties: {
+          name: {
+            type: 'string',
+          },
+          credit_card: {
+            type: 'string',
+          },
+        },
+        required: ['name'],
+        dependencies: {
+          credit_card: {
+            properties: {
+              billing_address: { type: 'string' },
+            },
+            required: ['billing_address'],
+          },
+        },
+      };
+      const schema = new Schema(jsonSchema);
+      const expected = new Schema({
+        type: 'object',
+        properties: {
+          credit_card: {
+            properties: {
+              billing_address: { type: 'string' },
+            },
+            required: ['billing_address'],
+          },
+        },
+        'x-schema-private-render-additional-info': false,
+        'x-schema-private-render-type': false,
+      });
+
+      const result = SchemaHelpers.getDependentSchemas(schema);
+      expect(result).toEqual(expected);
+    });
+
+    test('should return dependent schemas (complex case)', () => {
+      const jsonSchema = {
+        type: 'object',
+        properties: {
+          name: {
+            type: 'string',
+          },
+          credit_card: {
+            type: 'string',
+          },
+        },
+        required: ['name'],
+        dependencies: {
+          credit_card: {
+            properties: {
+              billing_address: { type: 'string' },
+            },
+            required: ['billing_address'],
+          },
+          billing_address: {
+            properties: {
+              account: { type: 'string' },
+            },
+            required: ['account'],
+          },
+        },
+      };
```

Comment:
Maybe it is better the complex case is for nested dependencies? 🤔 Or does this test covers lines that are not tested by the simple one?